### PR TITLE
Search by chef or meals

### PR DESCRIPTION
**Pull request by gary-bains**

### CODE REVIEW

**Comment by ThinkTink** [comment]
Quick request @garys-io , can you handle the merge conflicts?

**Comment by gary-bains** [comment]
> Quick request @garys-io , can you handle the merge conflicts?

Hi, sure, I'll do that but I'm waiting on Megan to make a PR which'll resolve many of the merge issue as I have merged their branch into mine earlier.

File: client/src/components/ChefCard.js
Comment (ThinkTink) [inline client/src/components/ChefCard.js]:
```suggestion
```

File: client/src/pages/EditProfile.js
Comment (ThinkTink) [inline client/src/pages/EditProfile.js]:
```suggestion
```

File: client/src/pages/Meals.js
Comment (ThinkTink) [inline client/src/pages/Meals.js:48]: Do we need `[...cuisines]` everywhere instead of just `cuisines`?

File: client/src/pages/Meals.js
Comment (ThinkTink) [inline client/src/pages/Meals.js]: Is there a reason to pass each of these individuall vs just passing meal itself?

File: client/src/utils.js
Comment (ThinkTink) [inline client/src/utils.js]: While it's fine for testing, I don't think we should plan on console logging errors. Depending on how we want to handle it, I think we should either show an error at the component level, or just ignore errors of this nature

File: server/routes/search.js
Comment (ThinkTink) [inline server/routes/search.js]: What is the purpose of this step?

File: server/routes/search.js
Comment (ThinkTink) [inline server/routes/search.js]: Should this be `res.json(meals)`?

File: server/routes/search.js
Comment (ThinkTink) [inline server/routes/search.js:49]: This whole route is a bit hard to follow. I'm thinking it might be easier to read if you pull some of the work out into functions?

File: client/src/pages/ChefSearch.js
Comment (ThinkTink) [inline client/src/pages/ChefSearch.js]: Are we using this anywhere?

**Review by ThinkTink — COMMENTED** [review]
Only needs a little bit of clean up before merging, but I do want to wait to merge until Megan's upcoming PR is merged so we can make sure we don't merge anything we don't intend to

File: client/src/pages/Meals.js
Comment (gary-bains) [reply client/src/pages/Meals.js:48]: Yes, we need it to convert the set into a list

File: client/src/pages/Meals.js
Comment (gary-bains) [reply client/src/pages/Meals.js]: fixed it!

File: client/src/pages/ChefSearch.js
Comment (gary-bains) [reply client/src/pages/ChefSearch.js]: refactored it to just use chefs array and removed the rest of the object

File: client/src/utils.js
Comment (gary-bains) [reply client/src/utils.js]: Should I change it to `const consoleErrorNonAbortErrors = () => {}`

Comment (ThinkTink) [reply client/src/utils.js]: I think we can just remove that function altogether. It's just being used inside the try-catch, so we just leave the catch block empty except for a comment explaining that we don't want to notify the user of any errors there

Comment (gary-bains) [reply client/src/utils.js]: Fixed it!

File: server/routes/search.js
Comment (gary-bains) [reply server/routes/search.js:49]: I agree, I refactored it

Comment (ThinkTink) [reply server/routes/search.js:49]: This is much better!

**Review by ThinkTink — APPROVED** [review]
Thanks for the fixes! Just make sure you look through it after Megan's PR merges to make sure there's nothing left here that we don't want to keep.

**Pull request merged**
